examples/wire_read_tcp.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python
+import argparse
+import sys
+
+import dns.exception
+import dns.message
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Read sequence of DNS wire formats prefixed with 2-byte "
+        "length field - like from TCP socket - and print the messages. This "
+        "format is used e.g. by dnsperf -B option and dnsgen."
+    )
+    parser.add_argument("infile")
+    args = parser.parse_args()
+    ok_msgs = 0
+    bad_msgs = 0
+
+    with open(args.infile, "rb") as infile:
+        while True:
+            offset = infile.tell()
+            len_wire = infile.read(2)
+            if len(len_wire) == 0:  # end of file - expected
+                break
+            if len(len_wire) == 1:
+                raise ValueError("incomplete length preambule, offset", offset)
+            len_msg = int.from_bytes(len_wire, byteorder="big", signed=False)
+            print(f"; msg offset 0x{offset + 2:x}, length {len_msg} bytes")
+            msg_wire = infile.read(len_msg)
+            if len(msg_wire) != len_msg:
+                raise ValueError(
+                    "incomplete message, expected != got, length field offset",
+                    len_msg,
+                    len(msg_wire),
+                    offset,
```

Comment:
I might be missing something, but this prints an error like:
```
ValueError: ('incomplete message, expected != got, length field offset', 258, 1, 0)
```
instead of a string with interpolated values.  Is that intentional?

Reply:
I personally prefer to keep individual values as objects so higher-level code can work with it if needed, but I can see your point. I will amend the message.